quix-frontend/client/src/lib/ui/components/dropdown/Dropdown.tsx

```diff
@@ -1,53 +1,88 @@
-import React, { useState } from 'react';
+import React, { useRef, useState } from 'react';
 import ReactDOM from 'react-dom';
 import { usePopper } from 'react-popper';
 import { Placement } from '@popperjs/core';
+import { isNil } from 'lodash'
+import { useOutsideAlerter } from '../../../../services/hooks';
 
 interface DropdownProps {
-  element: JSX.Element,
-  OptionsWrapper: React.ComponentType<InjectedDropdownProps>,
-  options: JSX.Element[],
-  isOpen: boolean,
-  spanClass?: string,
+  toggle: JSX.Element,
+  options: any[],
+  isOpen?: boolean,
+  classes?: {
+    inputWrapper?: string;
+  },
   placement?: Placement,
+  children?(options: any[]): JSX.Element;
+  states?: {
+    toggle?: {
+      onClick?: boolean;
+      onKeyDown?: boolean;
+      onFocus?: boolean;
+    };
+    list?: {
+      onClick?: boolean;
+    };
+  }
 }
 
-interface InjectedDropdownProps {
-  options: JSX.Element[],
+const createDropdownHelperIfNotExist = () => {
+  if (!document.querySelector('.bi-dropdown-helper')) {
+    const element = document.createElement('div');
+    element.className = 'bi-dropdown-helper';
+    document.body.appendChild(element);
+  }
 }
 
 export const Dropdown = ({
-  element,
-  OptionsWrapper,
+  toggle,
   options,
   isOpen,
-  spanClass,
+  classes,
   placement,
+  children,
+  states,
 }: DropdownProps) => {
-  const [_referenceElement, setReferenceElement] = useState(null);
-  const [referenceOptions, setReferenceOptions] = useState(null);
-  const { styles, attributes } = usePopper(_referenceElement, referenceOptions, {
+  createDropdownHelperIfNotExist();
+  const isOpenDefined = !isNil(isOpen);
+
+  const [_isOpen, setIsOpen] = useState(false);
+  const referenceToggle = useRef(null);
+  const referenceOptions = useRef(null);
+
+  useOutsideAlerter([referenceToggle, referenceOptions], () => setIsOpen(false));
```

Comment:
lets give this hook a more meaningful name, right now it doesn't sound like anything related to dom element clicks.